torch/lib/c10d/FileStore.cpp

```diff
@@ -56,9 +56,16 @@ class Lock {
 
   Lock(const Lock& that) = delete;
 
+  Lock& operator=(Lock&& other) noexcept {
+    if (this != &other) {
+      fd_ = other.fd_;
+      other.fd_ = -1;
+    }
+    return *this;
+  }
+
   Lock(Lock&& other) noexcept {
-    fd_ = other.fd_;
-    other.fd_ = -1;
+    *this = std::move(other);
```

Comment:
what is the reason for treading move-assign and move-constor differently? (One using move and another manually assigned)

Reply:
> 
> 
> what is the reason for treading move-assign and move-constor differently? (One using move and another manually assigned)

？？I don't know what you mean. I modify the code to treat move assignment and move constor consistently.

Reply:
I mean for the following code in move-assign:

```c++
  Lock& operator=(Lock&& other) noexcept {
    if (this != &other) {
      fd_ = other.fd_;
      other.fd_ = -1;
    }
    return *this;
  }
```

Can the two assignments be replaced by a move to stay consistent with the move constructor?

Reply:
They are ints, I think using assignment directly is clear. Also it's unclear what is the value of other.fd_ when using std::move. Assigning -1 to it makes further system calls fail.